File: back/MinhasDespesas/MinhasDespesas.Models/Dtos/DespesaDto.cs
namespace MinhasDespesas.Models.Dtos;

public class DespesaDto
{
    public Guid DespesaId { get; set; }
    // public PeriodoDto Periodo { get; set; }
    public Guid PeriodoId { get; set; }
    public GrupoDespesaDto GrupoDespesa { get; set; }
    public Guid GrupoDespesaId { get; set; }
    public DateTime DataCadastro { get; set; } = DateTime.SpecifyKind(DateTime.Now, DateTimeKind.Utc);
    public DateTime? DataVencimento { get; set; }
    public DateTime? DataQuitacao { get; set; }
    public string Descricao { get; set; }
    public double Valor { get; set; }
    public double ValorPago { get; set; }
    public double ValorFalta { get; set; }
    public IList<DespesaPagamentoDto> Pagamentos { get; set; }
}

File: front/minhas-despesas/src/models/despesa-dto.ts
import { GrupoDespesaDto } from "./grupo-despesa-dto";
import { PeriodoDto } from "./periodo-dto";

export interface DespesaDto {
    despesaId: string;
    periodo?: PeriodoDto;
    grupoDespesa: GrupoDespesaDto;
    dataCadastro: Date;
    dataVencimento?: Date;
    dataQuitacao?: Date;
    descricao: string;
    valor: number;
    valorPago: number;
    valorFalta: number;
}